### ISSUE TITLE
Outputwords on valid2 does not make any sense.

### ISSUE THREAD

**Issue by qyouurcs** [s-gupta/visual-concepts#2]
Hello,

I am using the scripts_all and trying to output the visual attributes for valid2.

Currently, I am able to reproduce the results on split test using the pre-trained model with the commind:

`
python run_mil.py --task output_words --gpu 1 --model output/vgg/snapshot_iter_240000.caffemodel --test_set test --calibration_set valid1 --vocab_file vocabs/vocab_train.pkl
`
However, when I try to ourput words for valid2 with either 
`
python run_mil.py --task output_words --gpu 1 --model output/vgg/snapshot_iter_240000.caffemodel --test_set valid2 --calibration_set valid2 --vocab_file vocabs/vocab_train.pkl
`
or
`
python run_mil.py --task output_words --gpu 1 --model output/vgg/snapshot_iter_240000.caffemodel --test_set valid2 --calibration_set valid1 --vocab_file vocabs/vocab_train.pkl
`
I kind of produce the following results (for example using "prec" metric) for most of the images. 

```
112798: a (1.00), on (0.81), of (0.79), in (0.77), the (0.76), with (0.72), and (0.65), is (0.54), to (0.41), man (0.38), sitting (0.32), an (0.31), two (0.27), at (0.26), standing (0.25), next (0.25), are (0.24),

185838: a (1.00), on (0.81), of (0.79), in (0.77), the (0.76), with (0.72), and (0.65), is (0.54), to (0.41), man (0.38), sitting (0.32), an (0.31), two (0.27), at (0.26), standing (0.25), next (0.25), are (0.24),

519874: a (1.00), on (0.81), of (0.79), in (0.77), the (0.76), with (0.72), and (0.65), is (0.54), to (0.41), man (0.38), sitting (0.32), an (0.31), two (0.27), at (0.26), standing (0.25), next (0.25), are (0.24),

235319: a (1.00), on (0.81), of (0.79), in (0.77), the (0.76), with (0.72), and (0.65), is (0.54), to (0.41), man (0.38), sitting (0.32), an (0.31), two (0.27), at (0.26), standing (0.25), next (0.25), are (0.24),

420882: a (1.00), on (0.81), of (0.79), in (0.77), the (0.76), with (0.72), and (0.65), is (0.54), to (0.41), man (0.38), sitting (0.32), an (0.31), two (0.27), at (0.26), standing (0.25), next (0.25), are (0.24),

521071: a (1.00), on (0.81), of (0.79), in (0.77), the (0.76), with (0.72), and (0.65), is (0.54), to (0.41), man (0.38), sitting (0.32), an (0.31), two (0.27), at (0.26), standing (0.25), next (0.25), are (0.24),

322194: a (1.00), on (0.81), of (0.79), in (0.77), the (0.76), with (0.72), and (0.65), is (0.54), to (0.41), man (0.38), sitting (0.32), an (0.31), two (0.27), at (0.26), standing (0.25), next (0.25), are (0.24),
```

Does anyone have the same problem?

Thanks

**Comment by s-gupta** [comment]
Hi,

Thanks for pointing this out. This issue was because the precomputed scores for valid2 set were incomplete. I fixed this and uploaded a new tarball which has complete scores for valid1, valid2 and test sets. If you re-run now you should get the right output. 

You can also just re-compute the scores on valid2 set locally at your end. Look at https://github.com/s-gupta/visual-concepts/blob/master/scripts/script_all.py#L16-L21 for how to do it.

Thanks!
